public/app/features/alerting/unified/components/rule-editor/DashboardPicker.tsx

```diff
@@ -0,0 +1,195 @@
+
+  const [panelFilter, setPanelFilter] = useState('');
+
+  const { useSearchQuery, useDashboardQuery } = dashboardApi;
+
+  const { currentData: filteredDashboards = [], isFetching: isDashSearchFetching } = useSearchQuery({
+    query: debouncedDashboardFilter,
+  });
+  const { currentData: dashboardResult, isFetching: isDashboardFetching } = useDashboardQuery(
+    { uid: dashboardUid ?? '' },
+    { skip: !dashboardUid }
+  );
+
+  useDebounce(
+    () => {
+      setDebouncedDashboardFilter(dashboardFilter);
+    },
+    500,
+    [dashboardFilter]
+  );
+
+  const handleDashboardChange = (dashboardUid: string) => {
+    onDashboardChange(dashboardUid);
+    onPanelChange(undefined);
+  };
+
+  const filteredPanels =
+    dashboardResult?.dashboard?.panels
+      ?.filter((panel): panel is PanelDTO => panel.title?.includes(panelFilter))
+      .sort(panelSort) ?? [];
+
+  const DashboardRow = ({ index, style }: { index: number; style: CSSProperties }) => {
+    const dashboard = filteredDashboards[index];
+    const isSelected = dashboardUid === dashboard.uid;
+
+    return (
+      <div
+        title={dashboard.title}
+        style={style}
+        className={cx(styles.row, { [styles.rowOdd]: index % 2 === 1, [styles.rowSelected]: isSelected })}
+        onClick={() => handleDashboardChange(dashboard.uid)}
+      >
+        <div className={styles.dashboardTitle}>{dashboard.title}</div>
+        <div className={styles.dashboardFolder}>
+          <Icon name="folder" /> {dashboard.folderTitle}
+        </div>
+      </div>
+    );
+  };
+
+  const PanelRow = ({ index, style }: { index: number; style: CSSProperties }) => {
+    const panel = filteredPanels[index];
+    const isSelected = panelId === panel.id;
+
+    return (
+      <div
+        style={style}
+        className={cx(styles.row, { [styles.rowOdd]: index % 2 === 1, [styles.rowSelected]: isSelected })}
+        onClick={() => onPanelChange(panel.id)}
+      >
+        {panel.title || '<No title>'}
+      </div>
+    );
+  };
+
+  return (
+    <div className={styles.container}>
+      <FilterInput
+        value={dashboardFilter}
+        onChange={setDashboardFilter}
+        title="Search dashboard"
+        placeholder="Search dashboard"
+        autoFocus
+      />
+      <FilterInput value={panelFilter} onChange={setPanelFilter} title="Search panel" placeholder="Search panel" />
+
+      <div style={{ flex: '1 1 auto' }}>
+        {isDashSearchFetching && (
+          <LoadingPlaceholder text="Loading dashboards..." className={styles.loadingPlaceholder} />
+        )}
+
+        {!isDashSearchFetching && (
+          <AutoSizer>
+            {({ height, width }) => (
+              <FixedSizeList itemSize={50} height={height} width={width} itemCount={filteredDashboards.length}>
+                {DashboardRow}
+              </FixedSizeList>
+            )}
+          </AutoSizer>
+        )}
+      </div>
+
+      <div style={{ flex: '1 1 auto' }}>
```

Comment:
Should we create a class to hold this styles instead? to avoid inlining and also I see it repeated above.

Reply:
We have a `Stack` component available in `@grafana/experimental` that we can use for flex layouts